misc/tiktok_api.py

```
@@ -1,163 +1,540 @@
-        self.rapid_link = 'https://tokapi-mobile-version.p.rapidapi.com/v1/post'
-        self.rapid_headers = {
-            "X-RapidAPI-Key": config["api"]["rapid_token"],
-            "X-RapidAPI-Host": "tokapi-mobile-version.p.rapidapi.com"
-        }
-        self.video_info_params = {'minimal': 'false'}
-        self.mobile_regex = re.compile(r'https?:\/\/[^\s]+tiktok.com\/[^\s]+')
-        self.web_regex = re.compile(r'https?:\/\/www.tiktok.com\/@[^\s]+?\/video\/[0-9]+')
-        self.mus_regex = re.compile(r'https?://www.tiktok.com/music/[^\s]+')
+    _executor = ThreadPoolExecutor(max_workers=4)
+
+    def __init__(self, proxy: str = None):
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)